overlord/servicestate/quotas.go

```diff
@@ -29,21 +29,49 @@ import (
 
 var ErrQuotaNotFound = errors.New("quota not found")
 
+func verifyRequiredCGroupsEnabledSingle(group *quota.Group) error {
+	if group.MemoryLimit != 0 && memoryCGroupError != nil {
+		return memoryCGroupError
+	}
+	if group.CpuLimit != nil {
+		if len(group.CpuLimit.AllowedCpus) > 0 && cpuSetCGroupError != nil {
+			return cpuSetCGroupError
+		}
+		if (group.CpuLimit.Count != 0 || group.CpuLimit.Percentage != 0) && cpuCGroupError != nil {
+			return cpuCGroupError
+		}
+	}
+	if group.TaskLimit != 0 && cpuCGroupError != nil {
+		return cpuCGroupError
+	}
+	return nil
+}
+
+func verifyRequiredCGroupsEnabled(groups map[string]*quota.Group) error {
+	for _, group := range groups {
+		if err := verifyRequiredCGroupsEnabledSingle(group); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
@Meulengracht this function will still cause the regression I mentioned - the issue is that we need to access the state of quotas in the state.json for quota groups even when the needed cgroups for that group(s) are disabled.

As I briefly mentioned to @atesburak, there are two ways to solve this problem:
* Add a new variant of AllQuotas which doesn't fail/check what cgroups are enabled and use that from the functions we need to be robust in while continuing to use the checking version which does fail in daemon/api_quotas.go
* Adjust the functions which are called from other managers (mainly SnapServiceOptions and EnsureSnapAbsentFromQuota) such that they are robust against servicestate.AllQuotas returning errors

I think the important things here are 1) to make sure that the work making this robust doesn't get duplicated and 2) that we don't block on things which don't need to be blocked on (again remember that this regression was introduced only to provide a nice error message for the user)